contrib/win32/win32compat/ssh-agent/keyagent-request.c

```diff
@@ -31,18 +31,33 @@
 
 #include "agent.h"
 #include "agent-request.h"
+#include "config.h"
 #include <sddl.h>
+#ifdef ENABLE_PKCS11
+#include "ssh-pkcs11.h"
+#endif
 
 #pragma warning(push, 3)
 
 #define MAX_KEY_LENGTH 255
-#define MAX_VALUE_NAME 16383
+#define MAX_VALUE_NAME_LENGTH 16383
+#define MAX_VALUE_DATA_LENGTH 2048
```

Comment:
Can you provide reference to microsoft docs to suggest why we change the MAX_VALUE_DATA_LENGTH  from 16383 to 2048.

Reply:
We didn't change any values here. We only:
- Change macro name from `MAX_VALUE_NAME ` to `MAX_VALUE_NAME_LENGTH ` to follow naming convention; and
- Add `MAX_VALUE_DATA_LENGTH` (See link below to related discussion)

Documentation regarding limits is [here](https://docs.microsoft.com/en-us/windows/win32/sysinfo/registry-element-size-limits)
(This was also discussed [here](https://github.com/PowerShell/openssh-portable/pull/537#discussion_r767640732))